compiler/jsgen.nim

```diff
@@ -2136,6 +2136,11 @@ proc optionaLine(p: Rope): Rope =
   else:
     return p & tnl
 
+proc genAsync(prc: PSym): Rope =
+  var returnType = prc.ast.sons[3][0]
```

Comment:
3 == paramsPos ?

Reply:
I didn't know it, neat. Should we expose those constants in macros?

Reply:
Macros has a ``params`` accessor.